tools/regenerate-docs.ts

```diff
@@ -89,6 +99,36 @@ const updateRulesList = (
   ].join('\n');
 };
 
+/**
+ * Generate a rule doc header for a particular rule.
+ * @param description - rule description
+ * @param name - rule name
+ * @returns {string[]} - lines for new header including marker
+ */
+const generateRuleHeaderLines = (description, name) => {
+  const lines = [`# ${description} (\`${name}\`)`];
+
+  lines.push(...getRuleNoticeLines(name));
+
+  lines.push(END_RULE_HEADER_MARKER);
+
+  return lines;
+};
```

Comment:
We might as well just return an array here:

```suggestion
const generateRuleHeaderLines = (description, name): string[] => [
  `# ${description} (\`${name}\`)`,
  ...getRuleNoticeLines(name),
  END_RULE_HEADER_MARKER,
];
```

Reply:
Good point, fixed.